### PR TITLE
github action to post adapter code coverage summary

### PR DESCRIPTION
**Pull request by onkarvhanumante**
PR introduces github action script that,
- Runs `go tool cover` command 
- Uploads coverage summary on pull request. Summary includes,
  - function wise stats
  - link to access heat map coverage report

 image image image

### CODE REVIEW

**Comment by SyntaxNode** [comment]
Nice! Some usability questions:
- Are we able to link directly to the artifact instead of having the author / reviewer click through?
- Can the artifact be a single HTML file so it can be opened directly in the browser instead of needing to be first downloaded?

I would like to enhance the comment made by this tool to make it clear this is not a pass/fail thing - that we have no specific minimums for code coverage. It would be also a good place to remind folks that we do not expect tests covering marshal and umarshal errr test coverage.

File: .github/workflows/adapter-code-coverage.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
+name: Adapter code coverage
+on:
+  pull_request_target:
+    paths: ["adapters/*/*.go"]
```
Comment (gargcreation1992) [inline .github/workflows/adapter-code-coverage.yml:4]: Should we restrict this to just adapters or Do you think we should do it for the entire prebid-server codebase?

File: .github/workflows/adapter-code-coverage.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            return files.length == 0 ? "" : JSON.stringify(files);
+
+      - name: Run coverage tests
+        id: run_coverage
+        if: ${{ steps.get_directories.outputs.result }} != ""
+        run: |
+          directories=$(echo '${{ steps.get_directories.outputs.result }}' | jq -r '.[]')
+          go mod download
+
+          # create a temporary directory to store the coverage output
```
Comment (gargcreation1992) [inline .github/workflows/adapter-code-coverage.yml:48]: Just curious, where is this temp directory created?

File: .github/workflows/adapter-code-coverage.yml (hunk 1 above)
Comment (onkarvhanumante) [reply .github/workflows/adapter-code-coverage.yml:4]: for initial run lets limit it to just adapters

File: .github/workflows/adapter-code-coverage.yml (hunk 2 above)
Comment (onkarvhanumante) [reply .github/workflows/adapter-code-coverage.yml:48]: https://github.com/prebid/prebid-server/blob/60a7704c051d35f7015f0541e62556c45229c48c/.github/workflows/adapter-code-coverage.yml#L9

job is configured to run on ubuntu, ` /tmp/<some-folder>` will be used to created

**Comment by onkarvhanumante** [comment]
> Nice! Some usability questions:
>     - Are we able to link directly to the artifact instead of having the author / reviewer click through? 
>     - Can the artifact be a single HTML file so it can be opened directly in the browser instead of needing to be first downloaded?
> 
> 
> I would like to enhance the comment made by this tool to make it clear this is not a pass/fail thing - that we have no specific minimums for code coverage. It would be also a good place to remind folks that we do not expect tests covering marshal and umarshal errr test coverage.


[e347d8d](https://github.com/prebid/prebid-server/pull/3004/commits/e347d8dc6161bf4df912fe879d71be759dfe2770) makes changes to include link in comment for viewing report in browser. Also updated message to include note for not adding tests covering marshal and umarshal error paths.

Testing:
https://github.com/onkarvhanumante/prebid-server/pull/20#issuecomment-1669529360
image

**Review by Sonali-More-Xandr — APPROVED** [review]

**Pull request merged**